casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -65,15 +107,32 @@ object ConflictSetMerger {
     val rejectionTargetF = (dc: Branch) => dc.map(cost).sum
     val optimalRejection = getOptimalRejection(rejectionOptions, rejectionTargetF)
     val toMerge          = branches diff optimalRejection
-    val rejected         = lateSet ++ rejectedAsDependents ++ optimalRejection.flatten
 
     for {
-      r                               <- Stopwatch.duration(toMerge.toList.flatten.traverse(stateChanges).map(_.combineAll))
+      baseMergeableChRes <- toMerge.flatten
```

Comment:
`toMerge` is already with excluded `optimalRejection`. Did you want to use `branches` instead?

Reply:
Well. After some thought, I think it would make calculations simple to use toMerge. Otherwise, if `optimalRejection` is conflicted with `negativeOrOverflowRejection`, we have to recalculate `overflowOrNegativeRejections` from `toMerge`(or some other thing which I am not certain).

Let's say the most basic structure
```
          D2 (transfer 10 from A to B)                           D4 for(_ <- @"X") {Nil} | (transfer 20 from A to B)
          D1 (transfer 20 from A to B) | @"X"!(1)          D3 (transfer 60 from A to B)
              \                                                                       /
                       Base
```
User A,B,C,D got 100 balance at `Base`.
D1 is short for deploy1.
D1 and D2 are in the same block1(B1). And D3 and D4 are in the same block2(B2). And we are going to merge B1 and B2. 

Based on the comm event conflicting rejection rules, D4 and D1 are conflicted. Let's say `optimalRejection` is D4.
And based on the negative rejection rules, if we look at `branches`, D2 can be  be the `negativeRejctsion` which we can actually reject nothing at all here.